services/elasticache/pkg/resource/replication_group/custom_update_api.go

```diff
@@ -30,76 +30,146 @@ func (rm *resourceManager) CustomModifyReplicationGroup(
 	latest *resource,
 	diffReporter *ackcompare.Reporter,
 ) (*resource, error) {
-	for _, diff := range diffReporter.Differences {
-		switch diff.Path {
-		case "Spec.NumNodeGroups":
-			return rm.updateShardConfiguration(ctx, desired, latest, diffReporter)
-		case "Spec.ReplicasPerNodeGroup":
-			return rm.updateReplicaCount(ctx, desired, latest, diffReporter)
+	// Order of operations when diffs map to multiple updates APIs:
+	// 1. updateReplicaCount() is invoked Before updateShardConfiguration()
+	//	  because both accept availability zones, however the number of
+	//	  values depend on replica count.
+
+	// increase/decrease replica count
+	if diff, message := rm.replicaCountDifference(desired, latest); diff != 0 {
+		fmt.Print(message)
+		if diff > 0 {
+			return rm.increaseReplicaCount(ctx, desired, latest)
 		}
+		return rm.decreaseReplicaCount(ctx, desired, latest)
+	}
+
+	// increase/decrease shards
+	if rm.shardConfigurationsDiffer(desired, latest) {
+		return rm.updateShardConfiguration(ctx, desired, latest)
 	}
+
+	// no updates
 	return nil, nil
 }
 
-func (rm *resourceManager) updateShardConfiguration(
+// replicaConfigurationsDifference returns
+// +ve number if desired replica count is greater than latest replica count
+// -ve number if desired replica count is less than latest replica count
+// 0 otherwise
+func (rm *resourceManager) replicaCountDifference(
+	desired *resource,
+	latest *resource,
+) (int, string) {
+	desiredSpec := desired.ko.Spec
+	latestStatus := latest.ko.Status
+
+	if desiredSpec.ReplicasPerNodeGroup != nil {
```

Comment:
Please place a comment into this function that basically says that there are two ways of setting replica counts for NodeGroups in Elasticache ReplicationGroups. The first way is to have the same replica count for all node groups. In this case, the `Spec.ReplicasPerNodeGroup` field is set to a non-nil-value integer pointer. The second way is to set different replica counts per node group. In this case, the `Spec.NodeGroupConfiguration` field is set to a non-nil `NodeGroupConfiguration` struct which contains a `NodeGroups` slice of `NodeGroup` structs that each have a `ReplicatCount` non-nil-value integer pointer field that contains the number of replicas for that particular node group.

Reply:
Thanks for the suggestion. Updated source code and added code comment.